### ISSUE TITLE
Assertion failure in debug build of SimpleIncSSReachAlgorithm

### ISSUE THREAD

**Issue by meyerphi** [libAlgora/AlgoraDyn#1]
I tested the simple incremental dynamic reachability algorithm of Algora using the following minimal example with two vertices `s` and `v` and one arc `(s,v)`, which is first added and then removed:
```c++
#include <iostream>

#include "graph/graph.h"
#include "graph.incidencelist/incidencelistgraph.h"
#include "algorithm.reach/simpleincssreachalgorithm.h"
#include "algorithm.reach/dynamicssreachalgorithm.h"

int main() {
    Algora::DiGraph* graph = new Algora::IncidenceListGraph();
    Algora::DynamicSSReachAlgorithm* reach = new Algora::SimpleIncSSReachAlgorithm();

    Algora::Vertex* s = graph->addVertex();

    reach->setGraph(graph);
    reach->setSource(s);

    std::cout << "s reachable: " << reach->query(s) << std::endl;
    Algora::Vertex* v = graph->addVertex();
    std::cout << "v reachable: " << reach->query(v) << std::endl;
    Algora::Arc* sv = graph->addArc(s, v);
    std::cout << "v reachable: " << reach->query(v) << std::endl;
    graph->removeArc(sv);
    std::cout << "v reachable: " << reach->query(v) << std::endl;

    delete reach;
    delete graph;
    return 0;
}
```
Saving the file as `test.cpp`, compiling it using
```
g++ -std=c++17 -o test -IAlgoraCore/src -IAlgoraDyn/src test.cpp AlgoraDyn/build/Debug/libAlgoraDyn.a AlgoraCore/build/Debug/libAlgoraCore.a
```
and then running `./test` produces the following output:
```
s reachable: 1
v reachable: 0
v reachable: 1
graphsearch: ../../src/algorithm.reach/simpleincssreachalgorithm.cpp:699: virtual void Algora::SimpleIncSSReachAlgorithm::onArcRemove(Algora::Arc*): Assertion `data->verifyReachability()' failed
```

Enabling `DEBUG_SISSREACH` in `simpleincssreachalgorithm.cpp` (and fixing some compile errors) further produces the following output:
```
s reachable: 1
v reachable: Propagating R from Vertex0#0.
0

A new arc was added: (Vertex0#0, Vertex0#1)
Propagating R from Vertex0#1.
v reachable: 1

(Vertex0#0, Vertex0#1) is about to be removed
Propagating ? from Vertex0#1.
Maximum allowed unknown state ratio exceeded, 1 > 0, recomputing.
Propagating R from Vertex0#0.
Setting remaining vertex Vertex0#1 with unknown state unreachable.
Wrong #reachable: stored 2, real 1
graphsearch: /home/philipp/local/Algora/Example/lib/AlgoraDyn/src/algorithm.reach/simpleincssreachalgorithm.cpp:699: virtual void Algora::SimpleIncSSReachAlgorithm::onArcRemove(Algora::Arc*): Assertion `data->verifyReachability()' failed.
```

**Comment by kalyi** [comment]
Hi Philipp,

thanks for reporting. It should be fixed in the latest commit (20b3544aadc408b1a28bc41dfdbbe09ce423c62b).

Best,

Kathrin

**Comment by meyerphi** [comment]
Thanks for the quick answer, this solves the issue.
